test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -869,6 +869,62 @@ public async Task WriteMetadataDocumentPayload_MustEqual_WriteMetadataDocumentAs
             // Assert
             Assert.Equal(syncPayload, asyncPayload);
         }
+
+        [Fact]
+        public async Task WriteLargeMetadataDocumentPayload_MustEqual_WriteLargeMetadataDocumentAsyncPayload_ForJsonCsdl()
+        {
+            // Arrange
+            var contentType = "application/json";
+
+            // Act
+            for (int i = 0; i < 10; i++)
+            {
+                // Json CSDL generated synchronously
+                string syncPayload = this.WriteAndGetPayload(_largeEdmModel, contentType, omWriter =>
+                {
+                    omWriter.WriteMetadataDocument();
+                });
+
+                // Json CSDL generated asynchronously
+                string asyncPayload = await this.WriteAndGetPayloadWithAsyncYieldStreamAsync(_largeEdmModel, contentType, async omWriter =>
+                {
+                    await omWriter.WriteMetadataDocumentAsync();
+                });
+
+                // Assert
+                Assert.Equal(syncPayload, asyncPayload);
+            }
+        }
+
+        [Fact]
+        public async Task WriteLargeMetadataDocumentAsync_CalledMultipleTimes_WorksForJsonCsdl_NoExceptionThrown()
+        {
+            // Arrange
+            string contentType = "application/json";
+
+            var message = new InMemoryMessage() { Stream = new AsyncYieldStream(new MemoryStream()) };
+
+            message.SetHeader("Content-Type", contentType);
+
+            var writerSettings = new ODataMessageWriterSettings();
+            writerSettings.EnableMessageStreamDisposal = false;
+            writerSettings.BaseUri = new Uri("http://www.example.com/");
+            writerSettings.SetServiceDocumentUri(new Uri("http://www.example.com/"));
+
+            // Act
+            for (int i = 0; i < 10; i++)
+            {
+                var exception = await Record.ExceptionAsync(async () =>

```

Comment:
@gathogojr That's correct. `Record.ExceptionAsync` will record any exception thrown by the given task. It will return the exception thrown or otherwise null.

In this case, I was not expecting any exception to be thrown and therefore used `Record.ExceptionAsync`.

Reply:
Where is `Record` defined? What does it do?

Reply:
It comes with `Xunit` library.